src/App.js

```diff
@@ -1,19 +1,45 @@
 import React from 'react';
 import './App.css';
 import chatMessages from './data/messages.json';
+import TimeStamp from './components/TimeStamp';
+import ChatEntry from './components/ChatEntry';
+import ChatLog from './components/ChatLog';
+import { useState } from 'react';
 
 const App = () => {
+  const [messages, setMessages] = useState(chatMessages);
+
+  const handleLike = (id) => {
+    const updatedMessages = messages.map((message) => {
```

Comment:
This works great in this case though I'd recommend using a callback with the update function because it will give you access to the most current previous state, which will ensure you're updating the correct value(s). To use that approach, pass a callback to the update function instead of a value. React will invoke that callback with an argument that represents the most recent state. Then you can use that in whatever computation you need and return the new value you want to set. e.g.
```js
const handleLike = (id) => {
  setMessages((previousMessages) => {
    return previousMessages.map((message) => {
      if (message.id === id) {
        return {
          ...message,
          liked: !message.liked,
        }
      }
    });
  });
};
```